experimental/css-has-pseudo/src/encode/test.mjs

```diff
@@ -130,201 +130,26 @@ testExtract(
 );
 
 testExtract(
-	'.a, [\\.x\\:has\\(\\%3E\\%20\\.b\\)][\\.x\\:has\\(\\%3E\\%20\\.b\\)]',
-	['.x:has(> .b)'],
+	'.a, [csstools-has-1m-2w-2p-37-14-1q-w-2j-2u-33-33-1p-y-37-33-31-2t-y-2l-15]',
+	[':has(> [foo="some"])'],
 );
 
 testExtract(
-	'[\\.x\\:has\\(\\%3E\\%20\\.b\\)][\\.x\\:has\\(\\%3E\\%20\\.b\\)], a',
-	['.x:has(> .b)'],
+	'[csstools-has-1m-2w-2p-37-14-1q-w-2j-2u-33-33-1p-y-37-33-31-2t-y-2l-15], a',
+	[':has(> [foo="some"])'],
 );
 
 testExtract(
-	'[\\.x\\:has\\(\\%3E\\%20\\.a\\)\\%20\\.b][\\.x\\:has\\(\\%3E\\%20\\.a\\)\\%20\\.b][\\.x\\:has\\(\\%3E\\%20\\.a\\)\\%20\\.b]',
-	['.x:has(> .a) .b'],
+	'[csstools-has-1m-2w-2p-37-14-1q-w-2j-2u-33-33-1p-y-37-33-31-2t-y-2l-15][csstools-has-1m-2w-2p-37-14-1q-w-2j-2u-33-33-1p-y-37-33-31-2t-y-2l-15][csstools-has-1m-2w-2p-37-14-1q-w-2j-2u-33-33-1p-y-37-33-31-2t-y-2l-15]',
+	[':has(> [foo="some"])'],
 );
 
 testExtract(
-	'[\\.x\\:has\\(\\%3E\\%20\\.\\%F0\\%9F\\%A7\\%91\\%F0\\%9F\\%8F\\%BE\\%E2\\%80\\%8D\\%F0\\%9F\\%8E\\%A4\\)][\\.x\\:has\\(\\%3E\\%20\\.\\%F0\\%9F\\%A7\\%91\\%F0\\%9F\\%8F\\%BE\\%E2\\%80\\%8D\\%F0\\%9F\\%8E\\%A4\\)]',
+	'[' + encodeCSS('.x:has(> .🧑🏾‍🎤)') +']',
 	['.x:has(> .🧑🏾‍🎤)'],
 );
 
 testExtract(
-	'[\\.x\\:has\\(\\%3E\\%20\\.a\\:has\\(\\%20\\%2B\\%20\\.b\\)\\)][\\.x\\:has\\(\\%3E\\%20\\.a\\:has\\(\\%20\\%2B\\%20\\.b\\)\\)][\\.x\\:has\\(\\%3E\\%20\\.a\\:has\\(\\%20\\%2B\\%20\\.b\\)\\)]',
+	'[' + encodeCSS('.x:has(> .a:has( + .b))') + ']',
 	['.x:has(> .a:has( + .b))'],
 );
-
```

Comment:
The previous encoder/decoder had to cover escaped `[`/`]`.
This required a lot of tests.

The new encoder/decoder doesn't have this complexity so a lot of tests became unneeded.